File: back_end/MindEarth/MindEarth.Web/Features/Roles/DTO/DTO_Menu.cs
using MindEarth.Database.Entity;

namespace MindEarth.Web.Features.Roles.DTO
{
    public class DTO_Menu
    {

        public string RowId { get; set; } = null!;

        public string Text { get; set; } = null!;


        public bool IsActive { get; set; }

        public List<DTO_MenuDetail> MenuDetails { get; set; } = new List<DTO_MenuDetail>();

    }

    public class DTO_MenuDetail
    {

        public string MenuDetailId { get; set; } = null!;
        public string Text { get; set; } = null!;
        public bool IsActive { get; set; }
        public bool AddRight { get; set; }
        public bool ViewRight { get; set; }
        public bool ModifyRight { get; set; }
        public bool DeleteRight { get; set; }
    }
}

File: admin_ui/report_admin_ui/src/app/features/auth/user-controller/models/dto-menu.ts
export class DtoMenu {
    rowId?: string;
    text?: string;
    isActive?: boolean;
    menuDetails?: DTO_MenuDetail[];

}

export class DTO_MenuDetail{
    rowId?: string;
    text?: string;
    isActive?: boolean;
    addRight?: boolean;
    viewRight?: boolean;
    modifyRight?: boolean;
    deleteRight?: boolean;
    menuDetailId?:string;
}